utils/handles/socket.c

```diff
@@ -74,7 +74,12 @@ int tls_socket_send(mbedtls_ctx_t *ctx, char const *data, size_t size) {
   return mbedtls_ssl_write(&ctx->ssl, (const unsigned char *)data, size);
 }
 
-int tls_socket_recv(mbedtls_ctx_t *ctx, char *data, size_t size, uint64_t timeout) {
+int tls_socket_recv(mbedtls_ctx_t *ctx, char *data, size_t size, uint32_t timeout) {
+  if (timeout != 0) {
+    mbedtls_ssl_config *ssl_conf = (mbedtls_ssl_config *)malloc(sizeof(mbedtls_ssl_config));
```

Comment:
Your implementation is not matched with the doc: https://tls.mbed.org/api/ssl_8h.html#a0d925033b3a46a48b3f8acc1d743af90
please check the usage of `timeout` configuration in the document and make sure timeout is working properly.

Reply:
Thank you for pointing out my error. I will try to fx it up.